packages/load/src/schema/from-git.ts

```diff
@@ -0,0 +1,52 @@
+import * as simplegit from 'simple-git/promise';
+import {buildSchema} from 'graphql';
+
+import {SchemaHandler} from './loader';
+
+function isGit(pointer: string): boolean {
+  return pointer.toLowerCase().startsWith('git:');
+}
+
+// git:branch:path/to/file
+function extractData(
+  pointer: string,
+): {
+  ref: string;
+  path: string;
+} {
+  const parts = pointer.replace(/^git\:/i, '').split(':');
+
+  if (!parts || parts.length !== 2) {
+    throw new Error(
+      'Schema pointer should match "git:branchName:path/to/file"',
+    );
+  }
+
+  return {
+    ref: parts[0],
+    path: parts[1],
+  };
+}
+
+export const fromGit: SchemaHandler = function fromGit(pointer) {
+  if (isGit(pointer)) {
+    return async function load() {
+      const {ref, path} = extractData(pointer);
+      const git = simplegit();
+
+      let schemaString: string;
+
+      try {
+        schemaString = await git.show([`${ref}:${path}`]);
+      } catch (error) {
+        throw new Error('Unable to load schema from git: ' + error);
+      }
+
+      if (/\.(gql|graphql)$/i.test(path)) {
```

Comment:
Some extensions are missing here, it should match the extensions listed like here: https://github.com/kamilkisiela/graphql-inspector/blob/cd5af71aa693d21671d26845292d04e81ee0b67e/packages/cli/src/commands/introspect.ts#L31-L34